response_automater/types/streaming_request.py

```diff
@@ -0,0 +1,30 @@
+from pydantic import BaseModel, Field
+from pydantic.functional_validators import AfterValidator
+from typing_extensions import Annotated
+
+
+# TODO: Do we need to keep this check? If so would it be better to just check string length rather than calling split() ?
+def not_over_one_thousand_words(v: str):
+    assert len(v.split(" ")) <= 1000, "Question exceeded 1000 words"
+    return v
```

Comment:
We do need to have this check because there's a limit to the number of tokens that can be present in the API's embedding request and later on in the chat completion request body.

The number of allowed tokens depends on the model that's been configured in the settings, and it's not as straightforward as one word == one token. 75 words is approximately 100 tokens.

There's an explanation of how we could work out the number of tokens in the question in https://cookbook.openai.com/examples/how_to_count_tokens_with_tiktoken.

Here we could make sure that the question length isn't going to break the embeddings call where for the model we are using (text-embedding-ada-002) the limit is 8192 tokens, but it's going to have to be a pretty big question to break that limit!

This is more relevant to the Chat completion call where we use the `gpt-4` model because the limit will include the system prompt as well as the question, context, and ideally some conversation history too so that the memory is retained.

For now, maybe we set a lower word limit and change the comment to an explanation?